packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx

```diff
@@ -57,9 +54,8 @@ const SignupConfirmCodeContainer = ({
 } & RouteComponentProps) => {
   const authClient = useAuthClient();
   const sensitiveDataClient = useSensitiveDataClient();
-  const sensitiveData = sensitiveDataClient.getData(AUTH_DATA_KEY);
   const { keyFetchToken, unwrapBKey } =
-    (sensitiveData as SensitiveDataClientAuthKeys) || {};
+    sensitiveDataClient.getDataType(SensitiveData.Key.Auth) || {};
 
   const { oAuthKeysCheckError } = useOAuthKeysCheck(
```

Comment:
On reload, missing auth keys causes a Bad Request error message (not a great user experience). Currently there is a [comment in the oauth hooks file](https://github.com/mozilla/fxa/blob/fb2c2fbd48d4811c3e3efd52cf4bbaf95f0fa4f3/packages/fxa-settings/src/lib/oauth/hooks.tsx#L288-L296) indicating that this would be looked at in FXA-10707, but that requirement was not actually noted in the ticket. You've already gone above and beyond by updating the types safety more generally, so I'd agree this should be a separate ticket that looks at all uses of the `useOAuthKeysCheck`. We should file a ticket to get that done and update the comment so that it no longer references this PR.

Reply:
Right, I think that's fair to move this to another ticket. The real value from this patch is that we now can see the nullability immediately. So figuring out how to handle this case will be easier to reason in a follow-up.

That said, I tried this quick hack and it seems to work fine for sync and an RP sign up flow where I refresh on the confirm_signup_code page and I am able to continue the flow. Will have to verify this in another patch before I put it up.
```diff
diff --git a/packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx b/packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx
index 53c9b6fe0c..9fe4ae9a83 100644
--- a/packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx
+++ b/packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx
@@ -4,7 +4,7 @@
 
 import React, { useCallback, useEffect } from 'react';
 import { RouteComponentProps, useLocation } from '@reach/router';
-import { useNavigateWithQuery as useNavigate } from '../../../lib/hooks/useNavigateWithQuery';
+import { useNavigateWithQuery as useNavigate, useNavigateWithQuery } from '../../../lib/hooks/useNavigateWithQuery';
 import { currentAccount } from '../../../lib/cache';
 import {
   useFinishOAuthFlowHandler,
@@ -54,6 +54,7 @@ const SignupConfirmCodeContainer = ({
 } & RouteComponentProps) => {
   const authClient = useAuthClient();
   const sensitiveDataClient = useSensitiveDataClient();
+  const navigateWithQuery = useNavigateWithQuery();
   const { keyFetchToken, unwrapBKey } =
     sensitiveDataClient.getDataType(SensitiveData.Key.Auth) || {};
 
@@ -63,6 +64,10 @@ const SignupConfirmCodeContainer = ({
     unwrapBKey
   );
 
+  if (integration.isSync() && (!keyFetchToken || !unwrapBKey)) {
+    navigateWithQuery('/signin');
+  }
+
   const location = useLocation() as ReturnType<typeof useLocation> & {
     state: LocationState;
   };
```